contracts/DefaultStrategy.sol

```diff
@@ -522,248 +436,288 @@ contract DefaultStrategy is UUPSOwnableUpgradeable, UsingRegistryUpgradeable, Ma
+        public
+        view
+        returns (uint256 expectedStCelo, uint256 actualStCelo)
     {
-        remainingWithdrawal = withdrawal;
-        uint256 numberDeprecatedGroups = deprecatedGroups.length();
-        deprecatedGroupsWithdrawn = new address[](numberDeprecatedGroups);
-        deprecatedWithdrawalsPerGroup = new uint256[](numberDeprecatedGroups);
-        numberDeprecatedGroupsWithdrawn = 0;
-
-        for (uint256 i = 0; i < numberDeprecatedGroups; i++) {
-            numberDeprecatedGroupsWithdrawn++;
-            deprecatedGroupsWithdrawn[i] = deprecatedGroups.at(i);
-            uint256 currentStCeloInStrategy = defaultStrategyTotalStCeloVotes[
-                deprecatedGroupsWithdrawn[i]
-            ];
-            uint256 currentVotes = IManager(manager).toCelo(currentStCeloInStrategy);
-            deprecatedWithdrawalsPerGroup[i] = Math.min(remainingWithdrawal, currentVotes);
-            remainingWithdrawal -= deprecatedWithdrawalsPerGroup[i];
-
-            if (currentVotes == deprecatedWithdrawalsPerGroup[i]) {
-                if (!deprecatedGroups.remove(deprecatedGroupsWithdrawn[i])) {
-                    revert FailedToRemoveDeprecatedGroup(deprecatedGroupsWithdrawn[i]);
-                }
-                emit GroupRemoved(deprecatedGroupsWithdrawn[i]);
-            }
+        address head = activeGroups.getHead();
+        expectedStCelo = totalStCeloInDefaultStrategy / activeGroups.getNumElements();
+        if (group == head) {
+            uint256 divisionResidue = totalStCeloInDefaultStrategy -
+                (expectedStCelo * activeGroups.getNumElements());
+            expectedStCelo += divisionResidue;
+        }
 
-            subtractFromStrategyTotalStCeloVotes(
-                deprecatedGroupsWithdrawn[i],
-                currentStCeloInStrategy
-            );
+        actualStCelo = stCELOInGroup[group];
+    }
 
-            if (remainingWithdrawal == 0) {
-                break;
-            }
+    /**
+     * @notice Adds/substracts value to totals of group and
+     * total stCELO in default strategy.
+     * @param group The validator group that we are updating.
+     * @param stCeloAmount The amount of stCELO.
+     * @param add Whether to add or substract.
+     */
+    function updateGroupStCelo(
+        address group,
+        uint256 stCeloAmount,
+        bool add
+    ) internal {
+        if (add) {
+            stCELOInGroup[group] += stCeloAmount;
+            totalStCeloInDefaultStrategy += stCeloAmount;
+        } else {
+            stCELOInGroup[group] -= stCeloAmount;
+            totalStCeloInDefaultStrategy -= stCeloAmount;
         }
+    }
 
-        return (
-            deprecatedGroupsWithdrawn,
-            deprecatedWithdrawalsPerGroup,
-            numberDeprecatedGroupsWithdrawn,
-            remainingWithdrawal
-        );
+    function getLesserAndGreaterOfActiveGroups(
+        address originalKey,
```

Comment:
Naming: just `group`? Since the key is just the group's address, and better to be explicit about what this is referring to.